bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -314,11 +369,50 @@ func (p *packer) encodeTuple(t Tuple, nested bool) {
 // call Pack when using a Tuple with a FoundationDB API function that requires a
 // key.
 func (t Tuple) Pack() []byte {
-	p := packer{buf: make([]byte, 0, 64)}
+	p := newPacker()
 	p.encodeTuple(t, false)
 	return p.buf
 }
 
+// PackWithVersionstamp packs the specified tuple into a key for versionstamp operations
+func (t Tuple) PackWithVersionstamp() ([]byte, error) {
+	hasVersionstamp, err := t.HasIncompleteVersionstamp()
+	if err != nil {
+		return nil, err
+	}
+
+	p := newPacker()
+	p.encodeTuple(t, false)
+
+	if hasVersionstamp {
+		var scratch [4]byte
+		binary.LittleEndian.PutUint32(scratch[:], uint32(p.versionstampPos))
+		p.putBytes(scratch[:])
```

Comment:
I believe that this only works if the API version is greater than or equal to 520. Prior to API version 520, the offset was two bytes rather than 4. I believe the minimum API version for the go bindings is 200, so either we need to bump the bindings' minimum API version to 520 or handle that case by writing only two bytes when the user specifies the older API version.

Reply:
👍 I'll make it backwards compatible.